[datadog_dashboard] Add `clustering_pattern_field_path` and `group_by` for ListStream Widget (#2869)
DataDog/terraform-provider-datadog: docs/resources/dashboard.md

Optional:

> - `clustering_pattern_field_path` (String) Specifies the field for logs pattern clustering. Usable only with logs_pattern_stream.

Reviewer: ```suggestion
- `clustering_pattern_field_path` (String) Specifies the field for logs pattern clustering. Can only be used with `logs_pattern_stream`.
```

Author: It seems Github have issues to see outdated comment for too big file but it is fixed

- `event_size` (String) Size of events displayed in widget. Required if `data_source` is `event_stream`. Valid values are `s`, `l`.

> - `group_by` (Block List) Group by configuration for the List Stream Widget. Group by can be used only with logs_pattern_stream (up to 4 items) or logs_transaction_stream (one group by item is required) list stream source. (see [below for nested schema](#nestedblock--widget--group_definition--widget--list_stream_definition--request--query--group_by))

Reviewer: ```suggestion
- `group_by` (Block List) Group by configuration for the List Stream widget. Group by can only be used with `logs_pattern_stream` (up to 4 items) or `logs_transaction_stream` (one group by item is required) list stream source. (see [below for nested schema](#nestedblock--widget--group_definition--widget--list_stream_definition--request--query--group_by))
```

Author: It seems Github have issues to see outdated comment for too big file but it is fixed

Optional:

> - `clustering_pattern_field_path` (String) Specifies the field for logs pattern clustering. Usable only with logs_pattern_stream.

Reviewer: ```suggestion
- `clustering_pattern_field_path` (String) Specifies the field for logs pattern clustering. Can only be used with `logs_pattern_stream`.
```

Author: It seems Github have issues to see outdated comment for too big file but it is fixed

- `event_size` (String) Size of events displayed in widget. Required if `data_source` is `event_stream`. Valid values are `s`, `l`.

> - `group_by` (Block List) Group by configuration for the List Stream Widget. Group by can be used only with logs_pattern_stream (up to 4 items) or logs_transaction_stream (one group by item is required) list stream source. (see [below for nested schema](#nestedblock--widget--list_stream_definition--request--query--group_by))

Reviewer: ```suggestion
- `group_by` (Block List) Group by configuration for the List Stream widget. Group by can only be used with `logs_pattern_stream` (up to 4 items) or `logs_transaction_stream` (one group by item is required) list stream source. (see [below for nested schema](#nestedblock--widget--list_stream_definition--request--query--group_by))
```

Author: It seems Github have issues to see outdated comment for too big file but it is fixed
